### ISSUE TITLE
Not compatible with Node.js 14

### ISSUE THREAD

**Issue by boogerlad** [uNetworking/uWebSockets.js#298]
```
node_modules/uWebSockets.js/uws.js:31
                throw new Error('This version of µWS is not compatible with your Node.js build:\n\n' + e.toString());
                ^

Error: This version of µWS is not compatible with your Node.js build:

Error: Cannot find module './uws_linux_x64_83.node'
Require stack:
node_modules/uWebSockets.js/uws.js
```
Works fine with Node.js 13

**Comment by alexhultman** [comment]
Yes you see Node.js 14 was released yesterday and I am working on it.

**Comment by alexhultman** [comment]
It runs now, but as per usual we have significant drops in performance with this new version:

**Node.js 12.15.0:**
Running 10s test @ http://localhost:9001
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    **96.15us**   33.36us   1.42ms   84.71%
    Req/Sec    51.66k     7.07k   69.29k    66.83%
  1038175 requests in 10.10s, 70.30MB read
Requests/sec: **102790.74**
Transfer/sec:      6.96MB

**Node.js 13.1.0:**
Running 10s test @ http://localhost:9001
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   269.75us    2.61ms  59.44ms   99.44%
    Req/Sec    50.95k     6.15k   63.07k    67.82%
  1023792 requests in 10.10s, 69.32MB read
Requests/sec: **101369.67**
Transfer/sec:      6.86MB

**Node.js 14.0.0:**
Running 10s test @ http://localhost:9001
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   110.34us   35.47us 436.00us   85.36%
    Req/Sec    45.04k     5.12k   55.53k    71.29%
  904947 requests in 10.10s, 61.27MB read
Requests/sec:  **89599.46**
Transfer/sec:      6.07MB

Remains to be seen if those drops are persistent or will be fixed in following patch releases. It was probably since v8 I saw this large of a drop in performance with one single Node.js version though, so it frightens me what they are doing to the project.

**Comment by zorro-fr24** [comment]
hopefully they used an unsigned int in their performance throttle - it will eventually drop so low it might wrap around to ludicrous speed again ( Node.js 37.0.0? )

**Comment by alexhultman** [comment]
The sad thing is that Node.js 0.8.0 from the initial months was actually quite rapid for being a JS runtime. But since that day numbers have fallen steadily as more and more clowns poke around in the code. Ryan's initial design was almost decent.

**Comment by zorro-fr24** [comment]
https://github.com/nodejs/node/issues/32226 kanske

**Comment by alexhultman** [comment]
Hard to tell, I have no ArrayBuffers in use for this test only String. Other tests that include ArrayBuffer show big losses, but not really anything more than the loss with String, so I don't think this is exactly the same. What is clear though, is that performance has been properly messed up with this release.

It could be ArrayBuffer related, I mean the only way forward with a JIT is to move around the cost and shove it under the mat. You can't really improve performance anything more than what we already did in the 80s - it's all just a matter of moving things around and marketing it differently. 

Look at V8's Ignition marketing - it's an interpreter because they came to the conclusion JITs are slow and take up a lot of CPU time to JIT, which overall slow things down because most code is not a single for loop of 5 billion iterations doing the exact same thing over and over, so they removed it and went back to interpreter as it has no JIT:ing cost.

It's just hilarious to follow - nothing has changed since the 1980s just different brands on the same old algorithms.
